web-console/src/views/ingestion-view/ingestion-view.tsx

```diff
@@ -278,24 +283,37 @@ ORDER BY "rank" DESC, "created_time" DESC`;
     });
 
     this.taskQueryManager = new QueryManager({
-      processQuery: async capabilities => {
-        if (capabilities.hasSql()) {
-          return await queryDruidSql({
-            query: IngestionView.TASK_SQL,
+        processQuery: async capabilities => {
+          if (this.state.preferOverlord) {
+            if (capabilities.hasOverlordAccess()) {
+              const resp = await Api.instance.get(`/druid/indexer/v1/tasks`);
+              return IngestionView.parseTasks(resp.data);
+            } else if (capabilities.hasSql()) {
+              return await queryDruidSql({
+                query: IngestionView.TASK_SQL,
+              });
+            } else {
+              throw new Error(`must have SQL or overlord access`);
+            }
+          } else {
+            if (capabilities.hasSql()) {
+              return await queryDruidSql({
+                query: IngestionView.TASK_SQL,
+              });
+            } else if (capabilities.hasOverlordAccess()) {
+              const resp = await Api.instance.get(`/druid/indexer/v1/tasks`);
+              return IngestionView.parseTasks(resp.data);
+            } else {
+              throw new Error(`must have SQL or overlord access`);
+            }
+          }
+        },
+        onStateChange: tasksState => {
+          this.setState({
+            tasksState,
           });
-        } else if (capabilities.hasOverlordAccess()) {
-          const resp = await Api.instance.get(`/druid/indexer/v1/tasks`);
-          return IngestionView.parseTasks(resp.data);
-        } else {
-          throw new Error(`must have SQL or overlord access`);
-        }
-      },
-      onStateChange: tasksState => {
-        this.setState({
-          tasksState,
-        });
-      },
-    });
+        },
+     });
   }
 
```

Comment:
Add these two static functions to support the above refactor.

```suggestion
  static fetchTasksViaSql = async () => {
    return await queryDruidSql({
      query: IngestionView.TASK_SQL,
    });
  };

  static fetchTasksViaOverlord = async () => {
    const resp = await Api.instance.get(`/druid/indexer/v1/tasks`);
    return IngestionView.parseTasks(resp.data);
  };

```